### PR TITLE
Fix Overview Page UI will crash Issue

### LINKED ISSUE #39564: [Core] Dashboard page empty
### What happened + What you expected to happen

Navigating to http://127.0.0.1:8265/ I see something quickly loading, then after a split second the page is empty. No dashboard or any elements. When inspecting the page I see `You need to enable JavaScript to run this app`. No browser (Chromium) addon installed at all.

### Versions / Dependencies

ray 2.6.4
python 3.9.18
manjaro

### Reproduction script
`ray dashboard ray.yaml`

ray.yaml:
```yaml
# A unique identifier for the head node and workers of this cluster.
cluster_name: test

# Running Ray in Docker images is optional (this docker section can be commented out).
# This executes all commands on all nodes in the docker container,
# and opens all the necessary ports to support the Ray cluster.
# Empty string means disabled. Assumes Docker is installed.
#docker:
#    image: "rayproject/ray-ml:latest-gpu" # You can change this to latest-cpu if you don't need GPU support and want a faster startup
#    # image: rayproject/ray:latest-gpu   # use this one if you don't need ML dependencies, it's faster to pull
#    container_name: "ray_container"
#    # If true, pulls latest version of image. Otherwise, `docker run` will only pull the image
#    # if no cached version is present.
#    pull_before_run: True
#    run_options:   # Extra options to pass into "docker run"
#        - --ulimit nofile=65536:65536

provider:
    type: local
    head_ip: 192.168.0.101
    # You may need to supply a public ip for the head node if you need
    # to run `ray up` from outside of the Ray cluster's network
    # (e.g. the cluster is in an AWS VPC and you're starting ray from your laptop)
    # This is useful when debugging the local node provider with cloud VMs.
    # external_head_ip: YOUR_HEAD_PUBLIC_IP
    worker_ips:
      - 192.168.0.106
      - 192.168.0.107
      - 192.168.0.108
      - 192.168.0.110
    # Optional when running automatic cluster management on prem. If you use a coordinator server,
    # then you can launch multiple autoscaling clusters on the same set of machines, and the coordinator
    # will assign individual nodes to clusters as needed.
    #    coordinator_address: "<host>:<port>"

# How Ray will authenticate with newly launched nodes.
auth:
    ssh_user: myuser
    # You can comment out `ssh_private_key` if the following machines don't need a private key for SSH access to the Ray
    # cluster:
    #   (1) The machine on which `ray up` is executed.
    #   (2) The head node of the Ray cluster.
    #
    # The machine that runs ray up executes SSH commands to set up the Ray head node. The Ray head node subsequently
    # executes SSH commands to set up the Ray worker nodes. When you run ray up, ssh credentials sitting on the ray up
    # machine are copied to the head node -- internally, the ssh key is added to the list of file mounts to rsync to head node.
    # ssh_private_key: ~/.ssh/id_rsa

# The minimum number of workers nodes to launch in addition to the head
# node. This number should be >= 0.
# Typically, min_workers == max_workers == len(worker_ips).
# This field is optional.
min_workers: 4

# The maximum number of workers nodes to launch in addition to the head node.
# This takes precedence over min_workers.
# Typically, min_workers == max_workers == len(worker_ips).
# This field is optional.
#max_workers: TYPICALLY_THE_NUMBER_OF_WORKER_IPS
# The default behavior for manually managed clusters is
# min_workers == max_workers == len(worker_ips),
# meaning that Ray is started on all available nodes of the cluster.
# For automatically managed clusters, max_workers is required and min_workers defaults to 0.

# The autoscaler will scale up the cluster faster with higher upscaling speed.
# E.g., if the task requires adding more nodes then autoscaler will gradually
# scale up the cluster in chunks of upscaling_speed*currently_running_nodes.
# This number should be > 0.
upscaling_speed: 1.0

idle_timeout_minutes: 5

# Files or directories to copy to the head and worker nodes. The format is a
# dictionary from REMOTE_PATH: LOCAL_PATH. E.g. you could save your conda env to an environment.yaml file, mount
# that directory to all nodes and call `conda -n my_env -f /path1/on/remote/machine/environment.yaml`. In this
# example paths on all nodes must be the same (so that conda can be called always with the same argument)
file_mounts: {
    "/mnt/ray/env.yaml": "env.yaml",
}

# Files or directories to copy from the head node to the worker nodes. The format is a
# list of paths. The same path on the head node will be copied to the worker node.
# This behavior is a subset of the file_mounts behavior. In the vast majority of cases
# you should just use file_mounts. Only use this if you know what you're doing!
cluster_synced_files: []

# Whether changes to directories in file_mounts or cluster_synced_files in the head node
# should sync to the worker node continuously
file_mounts_sync_continuously: False

# Patterns for files to exclude when running rsync up or rsync down
rsync_exclude:
    - "**/.git"
    - "**/.git/**"

# Pattern files to use for filtering out files when running rsync up or rsync down. The file is searched for
# in the source directory and recursively through all subdirectories. For example, if .gitignore is provided
# as a value, the behavior will match git's behavior for finding and using .gitignore files.
rsync_filter:
    - ".gitignore"

# List of commands that will be run before `setup_commands`. If docker is
# enabled, these commands will run outside the container and before docker
# is setup.
initialization_commands: []

# List of shell commands to run to set up each nodes.
setup_commands:
    # If we have e.g. conda dependencies stored in "/path1/on/local/machine/environment.yaml", we can prepare the
    # work environment on each worker by:
    #   1. making sure each worker has access to this file i.e. see the `file_mounts` section
    #   2. adding a command here that creates a new conda environment on each node or if the environment already exists,
    #     it updates it:
    #      conda env create -q -n my_venv -f /path1/on/local/machine/environment.yaml || conda env update -q -n my_venv -f /path1/on/local/machine/environment.yaml
    #
    # Ray developers:
    # you probably want to create a Docker image that
    # has your Ray repo pre-cloned. Then, you can replace the pip installs
    # below with a git checkout <your_sha> (and possibly a recompile).
    # To run the nightly version of ray (as opposed to the latest), either use a rayproject docker image
    # that has the "nightly" (e.g. "rayproject/ray-ml:nightly-gpu") or uncomment the following line:
    # - pip install -U "ray[default] @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl"
    - source ~/mambaforge-pypy3/etc/profile.d/conda.sh && mamba env update -f /mnt/ray/env.yaml --prune

# Custom commands that will be run on the head node after common setup.
head_setup_commands: []

# Custom commands that will be run on worker nodes after common setup.
worker_setup_commands: []

# Command to start ray on the head node. You don't need to change this.
head_start_ray_commands:
  # If we have e.g. conda dependencies, we could create on each node a conda environment (see `setup_commands` section).
  # In that case we'd have to activate that env on each node before running `ray`:
  # - conda activate my_venv && ray stop
  # - conda activate my_venv && ulimit -c unlimited && ray start --head --port=6379 --autoscaling-config=~/ray_bootstrap_config.yaml
    - source ~/mambaforge-pypy3/etc/profile.d/conda.sh && conda activate test && ray stop
    - source ~/mambaforge-pypy3/etc/profile.d/conda.sh && conda activate test && ulimit -c unlimited && ray start --head --disable-usage-stats --port=6379 --autoscaling-config=~/ray_bootstrap_config.yaml --system-config='{"automatic_object_spilling_enabled":true,"max_io_workers":8,"min_spilling_size":104857600,"object_spilling_config":"{\"type\":\"filesystem\",\"params\":{\"directory_path\":\"/mnt/ray/object_spilling\"}}"}'


# Command to start ray on worker nodes. You don't need to change this.
worker_start_ray_commands:
  # If we have e.g. conda dependencies, we could create on each node a conda environment (see `setup_commands` section).
  # In that case we'd have to activate that env on each node before running `ray`:
  # - conda activate my_venv && ray stop
  # - ray start --address=$RAY_HEAD_IP:6379
    - source ~/mambaforge-pypy3/etc/profile.d/conda.sh && conda activate test && ray stop
    - source ~/mambaforge-pypy3/etc/profile.d/conda.sh && conda activate test && ulimit -c unlimited && ray start --address=$RAY_HEAD_IP:6379 --disable-usage-stats
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by chaowanggg**
## Why are these changes needed?

Fix #39564, for an unknown reason, the backend router 'cluster_status' may return empty data without 'cluster_status'. However, the frontend may need 'cluster_status' to parse information and generate components, potentially leading to a UI crash.

The quick fix in the UI is to add protection for splitting 'cluster_status'.

We are aiming to address this issue and merge it in Ray 2.7.1.

## Results
Now, the UI is available even though cluster_status is undefined.
Screenshot 2023-10-03 at 3 04 07 PM

## Reproduce the bug
Screenshot 2023-10-03 at 2 55 28 PM

In Component OverviewPage, 

change  `const { cluster_status } = useRayStatus();` to

```
const cluster_status = {
    result: true,
    data: { clusterStatus: undefined },
    message: "",
  };
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by chaowanggg** [comment]
We have 6 components on the Overview page, and it appears that any error could lead to the crash of the Overview Page. Should we consider adding a React Error Boundary to protect the Overview Page? @alanwguo

**Comment by alanwguo** [comment]
> We have 6 components on the Overview page, and it appears that any error could lead to the crash of the Overview Page. Should we consider adding a React Error Boundary to protect the Overview Page? @alanwguo

yes, can we add error boundaries?

**Comment by alanwguo** [comment]
> > We have 6 components on the Overview page, and it appears that any error could lead to the crash of the Overview Page. Should we consider adding a React Error Boundary to protect the Overview Page? @alanwguo
> 
> yes, can we add error boundaries?

maybe we should do that as a follow-up for post 2.7.1

File: dashboard/client/src/pages/overview/OverviewPage.tsx [hunk 1]
```diff
@@ -51,8 +51,8 @@ const useStyles = makeStyles((theme) =>
 export const OverviewPage = () => {
   const classes = useStyles();
 
-  const { cluster_status } = useRayStatus();
-
+  // const { cluster_status } = useRayStatus();
+  const cluster_status = undefined;
```
Comment (alanwguo) [inline dashboard/client/src/pages/overview/OverviewPage.tsx:55]: revert this

**Review by alanwguo — CHANGES_REQUESTED** [review]

File: dashboard/client/src/pages/overview/OverviewPage.tsx (hunk 1 above)
Comment (chaowanggg) [reply dashboard/client/src/pages/overview/OverviewPage.tsx:55]: Ohh, this is not ready for review... Just try to reproduce the errors

**Comment by chaowanggg** [comment]
> > > We have 6 components on the Overview page, and it appears that any error could lead to the crash of the Overview Page. Should we consider adding a React Error Boundary to protect the Overview Page? @alanwguo
> > 
> > 
> > yes, can we add error boundaries?
> 
> maybe we should do that as a follow-up for post 2.7.1

Sounds good, just filed an issue

https://github.com/ray-project/ray/issues/40070

**Comment by rickyyx** [comment]
> Fix https://github.com/ray-project/ray/issues/39564, for an unknown reason, the backend router 'cluster_status' may return empty data without 'cluster_status'.

Can we actually root cause this?

**Comment by alanwguo** [comment]
We should root cause it, but we need to mitigate the issue where the entire UI crashes, preferably for ray 2.7.1

**Review by alanwguo — APPROVED** [review]

**Comment by rkooo567** [comment]
@rickyyx @chaowanggg can you guys create an issue to root cause it? @rickyyx is it unexpected the cluster status returns an empty string?

**Comment by chaowanggg** [comment]
Link the issue here https://github.com/ray-project/ray/issues/40076

**Comment by chaowanggg** [comment]
> @rickyyx @chaowanggg can you guys create an issue to root cause it? @rickyyx is it unexpected the cluster status returns an empty string?

Link the issue here https://github.com/ray-project/ray/issues/40076

**Pull request merged**